File: backend/models/Appointment.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema({
  doctor: { type: mongoose.Schema.Types.ObjectId, ref: 'Doctor', required: true },
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  date: { type: Date, required: true },
  time: { type: String, required: true },
  timeSlot: { type: String, enum: ['morning', 'evening'], required: true },
  symptoms: { type: String },
  status: { 
    type: String, 
    enum: ['pending', 'confirmed', 'completed', 'cancelled'], 
    default: 'pending' 
  },
  amount: { type: Number, default: 5100 },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Appointment', appointmentSchema);

File: flutter_app/lib/models/appointment.dart
class Appointment {
  final String? id;
  final String doctorId;
  final String userId;
  final DateTime date;
  final String time;
  final String timeSlot;
  final String symptoms;
  final double amount;
  final String? status;
  final DateTime? createdAt;

  Appointment({
    this.id,
    required this.doctorId,
    required this.userId,
    required this.date,
    required this.time,
    required this.timeSlot,
    required this.symptoms,
    required this.amount,
    this.status,
    this.createdAt,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id']?.toString(),
      doctorId: json['doctorId']?.toString() ?? '',
      userId: json['userId']?.toString() ?? '',
      date: json['date'] != null
          ? DateTime.parse(json['date'])
          : DateTime.now(),
      time: json['time'] ?? '',
      timeSlot: json['timeSlot'] ?? '',
      symptoms: json['symptoms'] ?? '',
      amount: (json['amount'] ?? 0).toDouble(),
      status: json['status'],
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'doctorId': doctorId,
      'userId': userId,
      'date': date.toIso8601String(),
      'time': time,
      'timeSlot': timeSlot,
      'symptoms': symptoms,
      'amount': amount,
      'status': status,
      'createdAt': createdAt?.toIso8601String(),
    };
  }
}
